js/lib/index.js

```diff
@@ -3,34 +3,41 @@ var {ElectroGrammarParser} = require('./ElectroGrammarParser');
 var {ElectroGrammarListener} = require('./ElectroGrammarListener');
 var antlr4 = require('antlr4');
 
-var KeyPrinter = function() {
+var ElectroGrammarToObjectListener = function() {
   ElectroGrammarListener.call(this);
+  this.obj = {};
   return this;
 };
 
-KeyPrinter.prototype = Object.create(ElectroGrammarListener.prototype);
-KeyPrinter.prototype.constructor = KeyPrinter;
+ElectroGrammarToObjectListener.prototype = Object.create(ElectroGrammarListener.prototype);
+ElectroGrammarToObjectListener.prototype.constructor = ElectroGrammarToObjectListener;
 
-KeyPrinter.prototype.exitCapacitance = function(ctx) {
-  var prefix = ctx
-    .CPREFIX()
-    .getText()
-    .toLowerCase();
-  if (prefix === 'micro') {
-    prefix = 'u';
-  } else {
-    prefix = prefix[0];
-  }
-  var number = Number(ctx.NUMBER().getText());
-  console.log('capacitance', `${number}${prefix}F`);
+ElectroGrammarToObjectListener.prototype.enterCapacitance = function(ctx) {
+	var cprefix_lookup = {'u': 10e-6, 'n': 10e-9, 'p': 10e-12};
+	var number = Number(ctx.NUMBER().getText());
+	var cprefix = cprefix_lookup[ctx.CPREFIX().getText()];
+	this.obj['capacitance'] = number * cprefix;
 };
 
-var input = '100 micro farad \n';
-var chars = new antlr4.InputStream(input);
-var lexer = new ElectroGrammarLexer(chars);
-var tokens = new antlr4.CommonTokenStream(lexer);
-var parser = new ElectroGrammarParser(tokens);
-parser.buildParseTrees = true;
-var tree = parser.capacitance();
-var printer = new KeyPrinter();
-antlr4.tree.ParseTreeWalker.DEFAULT.walk(printer, tree);
+var parse = function(input) {
+	var chars = new antlr4.InputStream(input);
+	var lexer = new ElectroGrammarLexer(chars);
+	var tokens = new antlr4.CommonTokenStream(lexer);
+	var parser = new ElectroGrammarParser(tokens);
+	parser.buildParseTrees = true;
+
+	try {
```

Comment:
Completely useless, haven't figured out how to detect errors yet

Reply:
I believe we need to create an `ErrorListener`. Do you have a copy of the Antlr4 book?